rotkehlchen/history/price.py

```diff
@@ -351,3 +353,35 @@ def query_multiple_prices(
         )
 
         return assets_price
+
+    @staticmethod
+    def query_uniswap_position_price(
+            pool_token: EvmToken,
+            pool_token_amount: FVal,
+            timestamp: Timestamp,
+    ) -> Price:
+        """Return the position price at the given timestamp"""
+        evm_inquirer = Inquirer.get_evm_manager(chain_id=pool_token.chain_id).node_inquirer
+        block_number = evm_inquirer.get_blocknumber_by_time(timestamp)
+        if pool_token.protocol == UNISWAP_PROTOCOL:
+            pool_price = PriceHistorian._uniswapv2.get_pool_price(
+                pool_addr=pool_token.evm_address,
+                chain_id=pool_token.chain_id,
+                block_identifier=block_number,
+            )
```

Comment:
isn't that pool price what you need already? Don't you just need to multiply the price per the amount of shares that you have?